src/main/java/nl/ordina/jobcrawler/service/MatchSkillsService.java

```diff
@@ -55,9 +57,22 @@ public void changeMatch(Vacancy vacancy) {
             }
         }
         log.info(vacancy.getTitle() + " matched with skills: " + matchedSkills.toString());
-        vacancy.setSkills(matchedSkills);
+        return matchedSkills;//vacancy.setSkills(matchedSkills);
     } 
 
+    // Rematch all vacancies with the skills in the skill table
+    public void relinkSkills() throws Exception {
+            skillService.deleteReferencesToSkills();
+            List<Vacancy> vacancies = vacancyService.findAll();
+            for (Vacancy vacancy: vacancies) {
+                Set<Skill> matchingSkills = findMatchingSkills(vacancy);
+                // not very fast -- could be improved
+                for(Skill skill : matchingSkills) {
+                    skillService.update(vacancy.getId(), skill);
+                }
+//                skillService.createMatchingSkillLinks(vacancy, matchingSkills);
+            }
+    }
 
 
     private void addStandardSkill(String aSkill) {
```

Comment:
This is for testing purposes, so it can be deleted or moved to test classes

Reply:
no, it is an aid to develop the application